packages/gitbook/src/components/AIActions/AIActions.tsx

```diff
@@ -77,61 +121,36 @@ export function CopyMarkdown(props: {
 }) {
     const { markdownPageUrl, type, isDefaultAction } = props;
     const language = useLanguage();
-    const { copied, setCopied, loading, setLoading } = useCopiedStore();
-    const timeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
-
-    // Close the dropdown menu manually after the copy button is clicked
-    const closeDropdownMenu = () => {
-        const dropdownMenu = document.querySelector('div[data-radix-popper-content-wrapper]');
+    const basePathname = usePathname();
+    const { copied, loading, markdown, pathname, setState, copyWithTimeout } = useCopiedStore();
 
-        // Cancel if no dropdown menu is open
-        if (!dropdownMenu) return;
-
-        // Dispatch on `document` so that the event is captured by Radix's
-        // dismissable-layer listener regardless of focus location.
-        document.dispatchEvent(new KeyboardEvent('keydown', { key: 'Escape', bubbles: true }));
-    };
+    // Clear cached markdown when navigating to a new page
+    useEffect(() => {
+        if (basePathname && pathname !== basePathname) {
+            setState({ markdown: null, pathname: basePathname });
+        }
+    }, [basePathname, pathname, setState]);
```

Comment:
This is not a good pattern; the better approach for this is to have your state be a map (page -> state).

It's better to think of state in a functional programming way and have purity; instead of imperative calls to sync things

Reply:
We don't really have to, the user is not going to navigate between 10k pages in the site. But if you are concerned about this, you can use a LRU map (like [https://www.npmjs.com/package/quick-lru](https://www.npmjs.com/package/quick-lru))